hkl/util.py

```diff
@@ -342,17 +344,31 @@ def run_orientation_info(run):
         A Bluesky run, from databroker v2, such as ``cat.v2[-1]``.
     """
     devices = {}
+
     try:
-        run_conf = run.primary.config
-        for device in sorted(run_conf):
-            conf = run_conf[device].read()
-            if f"{device}_orientation_attrs" in conf:
-                # fmt:off
-                devices[device] = {
-                    item[len(device)+1:]: conf[item].to_dict()["data"][0]
-                    for item in conf
-                }
-                # fmt:on
+        if databroker.__version__ >= "2.0":
+            for descriptor in run.primary.descriptors:
+                for device, configuration in descriptor.get("configuration", {}).items():
+                    conf = configuration.get("data", {})
+                    if f"{device}_orientation_attrs" in conf:
+                        # fmt:off
+                        devices[device] = {
+                            item[len(device) + 1:]: value
+                            for item, value in conf.items()
+                        }
```

Comment:
What's an example of the `conf.items()`? I am trying to understand why we need that complicated indexing - `len(device)+1:`

Reply:
`device` is the name of the diffractometer geometry.  It is possible for more than one diffractometer geometry to be saved in a run's orientation info.  The keys start with `f"{device}_{thing}"` where `thing` is some parameter of the diffractometer's saved orientation.  The indexing strips away `f"{device}_"` for each device's orientation dictionary.

FWIW, this code does the brute force work that is more easily done by JSON or YAML handling.  But a switch to the newer interface developed recently would result in a breaking change now.  Since this is part of a minor version update, such a breaking change is out of the question.

Reply:
if this snippet works, then the intent might be clearer. 

```suggestion
                        devices[device] = {
                            item[len(f"{device}_"):]: value
                            for item, value in conf.items()
                        }
```

alternatively 
len(device + “_”)